File: src/backend/blog-sec/src/main/java/com/msi/blogsec/data/Post.java
package com.msi.blogsec.data;

import com.msi.blogsec.domain.constants.PostStatus;
import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;
import java.util.Set;

/**
 * @author : goliathhagar
 * @CreatedBy : IntelliJ IDEA
 * @ProjectName : blog-sec
 * @mailto : goliathhagar@outlook.com
 * @created : 3/21/23, Tuesday, 10:48 AM
 **/


@Builder
@NoArgsConstructor
@AllArgsConstructor
@Setter
@Getter
@Entity(name = "posts")
public class Post {
    @Id private String id;
    @ManyToOne(fetch = FetchType.EAGER, cascade = CascadeType.ALL) private Author author;
    private String title;
    private String content;
    @Enumerated(EnumType.STRING) private PostStatus status;
    private boolean commentAllowed;
    private String tags;
    @OneToMany(mappedBy = "post", fetch = FetchType.LAZY) private Set<Comment> comments;
    private LocalDateTime publishedAt;
    private LocalDateTime updatedAt;
    private LocalDateTime createdAt;
}

File: src/frontend/blog_sec_app/lib/data/post.dart
class Post {
  String? author;
  String? title;
  String? content;
  String? status;
  bool? commentAllowed;
  String? tags;
  String? publishedAt;
  String? updatedAt;
  String? createdAt;
  Links? lLinks;

  Post(
      {this.author,
      this.title,
      this.content,
      this.status,
      this.commentAllowed,
      this.tags,
      this.publishedAt,
      this.updatedAt,
      this.createdAt,
      this.lLinks});

  Post.fromJson(Map<String, dynamic> json) {
    author = json['author'];
    title = json['title'];
    content = json['content'];
    status = json['status'];
    commentAllowed = json['commentAllowed'];
    tags = json['tags'];
    publishedAt = json['publishedAt'];
    updatedAt = json['updatedAt'];
    createdAt = json['createdAt'];
    lLinks = json['_links'] != null ? Links.fromJson(json['_links']) : null;
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{};
    data['author'] = author;
    data['title'] = title;
    data['content'] = content;
    data['status'] = status ?? "DRAFT";
    data['commentAllowed'] = commentAllowed;
    data['tags'] = tags ?? "DRAFT";
    data['publishedAt'] = publishedAt;
    data['updatedAt'] = updatedAt;
    data['createdAt'] = createdAt;

    return data;
  }
}

class Links {
  Self? self;
  Self? comments;

  Links({this.self, this.comments});

  Links.fromJson(Map<String, dynamic> json) {
    self = json['self'] != null ? Self.fromJson(json['self']) : null;
    comments =
        json['comments'] != null ? Self.fromJson(json['comments']) : null;
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{};
    if (self != null) {
      data['self'] = self!.toJson();
    }
    if (comments != null) {
      data['comments'] = comments!.toJson();
    }
    return data;
  }
}

class Self {
  String? href;

  Self({this.href});

  Self.fromJson(Map<String, dynamic> json) {
    href = json['href'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{};
    data['href'] = href;
    return data;
  }
}
